Code/GraphMol/hanoitest.cpp

```diff
@@ -0,0 +1,1353 @@
+    TEST_ASSERT(m);
+    MolOps::sanitizeMol(*m);
+    std::vector<unsigned int> atomRanks;
+    //std::cerr<<smi<<std::endl;
+    RDKit::Canon::chiralRankMolAtoms(*m,atomRanks);
+    //std::copy(atomRanks.begin(),atomRanks.end(),std::ostream_iterator<unsigned int>(std::cerr," "));
+    //std::cerr<<std::endl;
+    TEST_ASSERT(atomRanks[0]<atomRanks[5]);
+    TEST_ASSERT(atomRanks[1]<atomRanks[4]);
+  }
+
+  {
+    // are double bonds being handled correctly?
+    std::string smi="O=C[C@H](F)CO";
+    RWMol *m =SmilesToMol(smi,0,0);
+    TEST_ASSERT(m);
+    MolOps::sanitizeMol(*m);
+    std::vector<unsigned int> atomRanks;
+    //std::cerr<<smi<<std::endl;
+    RDKit::Canon::chiralRankMolAtoms(*m,atomRanks);
+    //std::copy(atomRanks.begin(),atomRanks.end(),std::ostream_iterator<unsigned int>(std::cerr," "));
+    //std::cerr<<std::endl;
+    TEST_ASSERT(atomRanks[0]>atomRanks[5]);
+    TEST_ASSERT(atomRanks[1]>atomRanks[4]);
+  }
+
+  {
+    // are double bonds being handled correctly?
+    std::string smi="CC[C@](C)(CF)C=O";
+    RWMol *m =SmilesToMol(smi,0,0);
+    TEST_ASSERT(m);
+    MolOps::sanitizeMol(*m);
+    std::vector<unsigned int> atomRanks;
+    //std::cerr<<smi<<std::endl;
+    RDKit::Canon::chiralRankMolAtoms(*m,atomRanks);
+    //std::copy(atomRanks.begin(),atomRanks.end(),std::ostream_iterator<unsigned int>(std::cerr," "));
+    //std::cerr<<std::endl;
+    TEST_ASSERT(atomRanks[4]>atomRanks[6]);
+    TEST_ASSERT(atomRanks[1]<atomRanks[4]);
+  }
+
+  BOOST_LOG(rdInfoLog) << "Finished" << std::endl;
+}
+
+
+void test10(){
+  BOOST_LOG(rdInfoLog) << "testing unique ranks in w.r.t. renumbering." << std::endl;
+  unsigned int i=0;
+  while(smis[i]!="EOS"){
+    std::string smiles=smis[i++];
+//    std::cerr<< ">>>Molecule: " << smiles << std::endl;
+    ROMol *m = SmilesToMol(smiles);
+    TEST_ASSERT(m);
+    MolOps::assignStereochemistry(*m,true);
+    _renumberTest2(m,smiles,1);
+    delete m;
+  }
+  BOOST_LOG(rdInfoLog) << "Finished" << std::endl;
+}
+
+
+void test11(){
+  BOOST_LOG(rdInfoLog) << "testing mol fragments." << std::endl;
+  {
+    std::string smi="C[C@H]([C@H](c1ccccc1)O)N2CCCCC2.C[C@@H]([C@H](c1ccccc1)O)N2CCCCC2";
+    ROMol *m = SmilesToMol(smi);
+    TEST_ASSERT(m);
+    std::vector<std::string> vfragsmi;
+    std::vector<std::vector<int> > frags;
+    unsigned int numFrag = MolOps::getMolFrags(*m,frags);
+    for(unsigned i=0;i<numFrag;++i){
+      std::string smii = MolFragmentToSmiles(*m,frags[i],0,0,0,true);
+      std::cout << smii << std::endl;
+      vfragsmi.push_back(smii);
+    }
+    std::string smi1 = MolToSmiles(*m,true);
+    delete m;
+
+    smi="C[C@@H]([C@H](c1ccccc1)O)N2CCCCC2.C[C@H]([C@H](c1ccccc1)O)N2CCCCC2";
+    m = SmilesToMol(smi);
+    TEST_ASSERT(m);
+    std::string smi2 = MolToSmiles(*m,true);
+    delete m;
+    std::cout << smi1 << "\n" << smi2 << std::endl;
+    TEST_ASSERT(smi1==smi2);
+
+  }
+  BOOST_LOG(rdInfoLog) << "Finished" << std::endl;
+}
+
+
+void test12(){
```

Comment:
This doesn't actually test anything, just runs the canonicalization, it should probably be a real test.
I also wonder if it shouldn't be in the ChemicalReactions tests instead of here.